SpringProject/src/test/java/com/gradproject2019/AuthUtilsTest.java

```diff
@@ -0,0 +1,59 @@
+package com.gradproject2019;
+
+import com.gradproject2019.users.service.UserServiceImpl;
+import com.gradproject2019.utils.AuthUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AuthUtilsTest {
+
+    @Test
+    public void shouldCheckPasswordIsValidAgainstRegex() {
+        Assert.assertFalse(AuthUtils.validate("1@Aa", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("@Password", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("1@password", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("1Password", UserServiceImpl.PASSWORD_PATTERN));
+
+        Assert.assertTrue(AuthUtils.validate("1@Password", UserServiceImpl.PASSWORD_PATTERN));
+
+    }
+
+    @Test
+    public void shouldCheckEmailIsValidAgainstRegex () {
+        Assert.assertTrue(AuthUtils.validate("Gracesophia!1@gmail.com", UserServiceImpl.EMAIL_PATTERN));
+
+        Assert.assertFalse(AuthUtils.validate("gracesophiagmail.com", UserServiceImpl.EMAIL_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("gracesophia@gmailcom", UserServiceImpl.EMAIL_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("gracesophia @gmailcom", UserServiceImpl.EMAIL_PATTERN));
+    }
+
+    @Test
+    public void shouldCheckUsernameNoSpaces() {
+        Assert.assertFalse(AuthUtils.validate("Sophia Grace", UserServiceImpl.USERNAME_PATTERN));
+        Assert.assertTrue(AuthUtils.validate("SophiaGrace", UserServiceImpl.USERNAME_PATTERN));
+    }
+
+    @Test
+    public void shouldCheckPasswordMatchesHashedPassword() {
+        //given
+        String password = "password123";
+
+        //when
+        String hashedPassword = AuthUtils.hash(password);
+
+        //then
+        Assert.assertTrue(AuthUtils.verifyHash(password, hashedPassword));
+    }
+
+    @Test
+    public void shouldCheckPasswordDoesNotMatchHashedPassword() {
+        //given
+        String password = "password123";
+
+        //when
+        String hashedPassword = AuthUtils.hash(password);
+
+        //then
+        Assert.assertFalse(AuthUtils.verifyHash("WrongPassword", hashedPassword));
```

Comment:
I think this assertion is returning a false positive - you're using another method in the same class you're testing to test this method

Actually on further inspection it turns out all these methods are doing is using a method from another library - if you're not adding any extra logic then do these tests even have to be here? (open question)